mslib/msui/multilayers.py

```diff
@@ -0,0 +1,519 @@
+            if item in self.layers_priority:
+                self.parent.listLayers.removeItemWidget(item, 1)
+                self.layers_priority.remove(item)
+                self.update_priority_selection()
+
+    def priority_changed(self, new_index):
+        """
+        Get called whenever the user changes a priority for a layer
+        Finds out the previous index and switches the layer position in self.layers_priority
+        """
+        active_layers = self.get_active_layers()
+        old_index = [i for i in range(1, len(active_layers) + 1)]
+        for layer in active_layers:
+            value = self.get_multilayer_priority(layer)
+            if value in old_index:
+                old_index.remove(value)
+        old_index = old_index[0] - 1
+
+        to_move = self.layers_priority.pop(old_index)
+        self.layers_priority.insert(new_index, to_move)
+        self.update_priority_selection()
+
+    def multilayer_doubleclicked(self, item, column):
+        """
+        Gets called whenever the user double clicks an entry in the multilayer list
+        Currently this is used to draw the double clicked layer
+        """
+        if item.childCount() == 0:
+            item.draw()
+
+    def right_clicked(self, pointer, testing=False):
+        """
+        Gets called whenever the user right clicks somewhere in the multilayer list
+        For now this is used to enable syncing
+        """
+        widget = self.parent.listLayers.itemAt(pointer)
+        if not widget or widget.childCount() > 0:
+            return
+
+        sync_menu = QtWidgets.QMenu()
+        sync_menu.setContentsMargins(10, 0, 10, 0)
+        is_synced = QtWidgets.QCheckBox("Synchronise")
+        is_synced.setChecked(widget.is_synced)
+        action = QtWidgets.QWidgetAction(sync_menu)
+        action.setDefaultWidget(is_synced)
+        sync_menu.addAction(action)
+        sync_menu.addSeparator()
+        info_label = QtWidgets.QLabel("")
+        action2 = QtWidgets.QWidgetAction(sync_menu)
+        action2.setDefaultWidget(info_label)
+        sync_menu.addAction(action2)
+
+        levels_before, itimes_before, vtimes_before, crs_before = self.get_multilayer_common_options()
+        levels, itimes, vtimes, crs = self.get_multilayer_common_options(widget)
+        lost_levels = len(levels) - len(levels_before)
+        lost_itimes = len(itimes) - len(itimes_before)
+        lost_vtimes = len(vtimes) - len(vtimes_before)
+
+        info = f"Common levels: {len(levels)} {f'({lost_levels})' if lost_levels < 0 else ''}\n"
+        info += f"Common init times: {len(itimes)} {f'({lost_itimes})' if lost_itimes < 0 else ''}\n"
+        info += f"Common valid times: {len(vtimes)} {f'({lost_vtimes})' if lost_vtimes < 0 else ''}"
+        info_label.setText(info)
+
+        def check_changed(state):
+            widget.is_synced = not widget.is_synced
+            if widget.is_synced:
+                widget.setText(0, widget.text(0) + " (synced)")
```

Comment:
Where should I see this info?

Reply:
Directly in the list. After a layer is synced, the name of the layer is appended by `(synced)`
This makes it also possible to search synced layers by filtering for `(synced)`